MPCD Taylor Couette Virtual Particle Filler for GPU import missing

### Description

With the new Taylor Couette Geometry (`ConcentricCylinders`) for MPCD, it appears that some import statements for the virtual particle filler are missing on the GPU in the `mpcd/module.cc` file. Additionally, there might be performance issues as well, which require further investigation. 

**Suggested fix:** 
Add the following import statements to `mpcd/module.cc` : 

```
void export_ConcentricCylindersGeometryFillerGPU(pybind11::module&); 
export_ConcentricCylindersGeometryFillerGPU(m);
```


### Script

```python
import numpy as np
import hoomd
import gsd, gsd.hoomd

density = 10  # density
L= 10 # box size

device = hoomd.device.GPU()

simulation = hoomd.Simulation(device=device, seed=1)

frame = gsd.hoomd.Frame()
frame.particles.types = ['A']
frame.configuration.box = [L,L,L, 0, 0, 0]
frame.particles.N = 0

simulation.create_state_from_snapshot(frame)

snapshot  = simulation.state.get_snapshot()

Nmpcd = int(np.round(density*L**3))

pos = np.random.uniform(-L/2.,+L/2.,size=(Nmpcd,3))
snapshot.mpcd.types = ['A','W']
snapshot.mpcd.N = len(pos)
snapshot.mpcd.position[:] = pos

simulation.state.set_snapshot(snapshot)

cylinders = hoomd.mpcd.geometry.ConcentricCylinders(inner_radius=1,
                                                        outer_radius=4)

stream = hoomd.mpcd.stream.BounceBack(period=20, geometry=cylinders)
collide = hoomd.mpcd.collide.StochasticRotationDynamics(period=20,angle=130,kT=1.0)
filler = hoomd.mpcd.fill.GeometryFiller(type="W",
                                             density=density,
                                             kT=1.0,
                                             geometry=cylinders)

integrator = hoomd.mpcd.Integrator(
    dt=0.005,
    streaming_method=stream,
    collision_method=collide,
    virtual_particle_fillers=[filler],
    mpcd_particle_sorter=hoomd.mpcd.tune.ParticleSorter(trigger=400))

simulation.operations.integrator = integrator

simulation.run(1)
```


### Input files

_No response_

### Output

```shell
Traceback (most recent call last):
  File "/projects/bbhc/statt/MPCD_TC_test/test.py", line 52, in <module>
    simulation.run(1)
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/simulation.py", line 555, in run
    self.operations._schedule()
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/operations.py", line 211, in _schedule
    self.integrator._attach(sim)
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/operation.py", line 315, in _attach
    self._attach_hook()
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/mpcd/integrate.py", line 241, in _attach_hook
    self._virtual_particle_fillers._sync(self._simulation,
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/data/syncedlist.py", line 246, in _sync
    raise err
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/data/syncedlist.py", line 242, in _sync
    self._register_item(item)
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/data/syncedlist.py", line 190, in _register_item
    value._attach(self._simulation)
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/operation.py", line 315, in _attach
    self._attach_hook()
  File "/u/statt/.conda/envs/utilities/lib/python3.10/site-packages/hoomd/mpcd/fill.py", line 152, in _attach_hook
    assert class_ is not None, ("Virtual particle filler for geometry "
AssertionError: Virtual particle filler for geometry not found
```


### Expected output

Nothing, simulation should run. It does work on the CPU.

### Platform

GPU

### Installation method

Compiled from source

### HOOMD-blue version

4.9.1 (trunk-minor, commit a86cbe76426)

### Python version

3.10

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -16,6 +16,9 @@ Change Log
   (`#1925 <https://github.com/glotzerlab/hoomd-blue/pull/1925>`__)
 * Mesh potentials return only one volume or area when ``ignore_type == True``
   (`#1928 <https://github.com/glotzerlab/hoomd-blue/pull/1928>`__)
+* Ensure correct methods are used by ``hoomd.mpcd.fill.VirtualParticleFiller`` for certain
+  geometries, improving performance on the GPU in these cases
+  (`#1934 <https://github.com/glotzerlab/hoomd-blue/pull/1934>`__).
 
 *Added*
 

--- a/hoomd/mpcd/RejectionVirtualParticleFiller.h
+++ b/hoomd/mpcd/RejectionVirtualParticleFiller.h
@@ -162,10 +162,8 @@ template<class Geometry> void export_RejectionVirtualParticleFiller(pybind11::mo
     namespace py = pybind11;
     const std::string name = Geometry::getName() + "GeometryFiller";
     py::class_<mpcd::RejectionVirtualParticleFiller<Geometry>,
-               std::shared_ptr<mpcd::RejectionVirtualParticleFiller<Geometry>>>(
-        m,
-        name.c_str(),
-        py::base<mpcd::VirtualParticleFiller>())
+               mpcd::VirtualParticleFiller,
+               std::shared_ptr<mpcd::RejectionVirtualParticleFiller<Geometry>>>(m, name.c_str())
         .def(pybind11::init<std::shared_ptr<SystemDefinition>,
                             const std::string&,
                             Scalar,

--- a/hoomd/mpcd/RejectionVirtualParticleFillerGPU.cuh
+++ b/hoomd/mpcd/RejectionVirtualParticleFillerGPU.cuh
@@ -38,7 +38,7 @@ struct draw_virtual_particles_args_t
                                   const Scalar _vel_factor,
                                   const unsigned int _type,
                                   const unsigned int _N_virt_max,
-                                  const unsigned int _timestep,
+                                  const uint64_t _timestep,
                                   const unsigned int _seed,
                                   const unsigned int _filler_id,
                                   const unsigned int _block_size)
@@ -58,7 +58,7 @@ struct draw_virtual_particles_args_t
     const Scalar vel_factor;
     const unsigned int type;
     const unsigned int N_virt_max;
-    const unsigned int timestep;
+    const uint64_t timestep;
     const unsigned int seed;
     const unsigned int filler_id;
     const unsigned int block_size;
@@ -126,7 +126,7 @@ __global__ void draw_virtual_particles(Scalar4* d_tmp_pos,
                                        const Scalar vel_factor,
                                        const unsigned int type,
                                        const unsigned int N_virt_max,
-                                       const unsigned int timestep,
+                                       const uint64_t timestep,
                                        const unsigned int seed,
                                        const unsigned int filler_id,
                                        const unsigned int block_size,

--- a/hoomd/mpcd/RejectionVirtualParticleFillerGPU.h
+++ b/hoomd/mpcd/RejectionVirtualParticleFillerGPU.h
@@ -54,7 +54,7 @@ class PYBIND11_EXPORT RejectionVirtualParticleFillerGPU
 
     protected:
     //! Fill the volume outside the confinement
-    virtual void fill(unsigned int timestep);
+    virtual void fill(uint64_t timestep);
 
     private:
     GPUArray<bool> m_keep_particles; // Track whether particles are in/out of bounds for geometry
@@ -64,8 +64,7 @@ class PYBIND11_EXPORT RejectionVirtualParticleFillerGPU
     std::shared_ptr<Autotuner<1>> m_tuner2; //!< Autotuner for particle tagging
     };
 
-template<class Geometry>
-void RejectionVirtualParticleFillerGPU<Geometry>::fill(unsigned int timestep)
+template<class Geometry> void RejectionVirtualParticleFillerGPU<Geometry>::fill(uint64_t timestep)
     {
     // Number of particles that we need to draw (constant)
     const BoxDim& box = this->m_pdata->getBox();
@@ -182,10 +181,8 @@ template<class Geometry> void export_RejectionVirtualParticleFillerGPU(pybind11:
     namespace py = pybind11;
     const std::string name = Geometry::getName() + "GeometryFillerGPU";
     py::class_<mpcd::RejectionVirtualParticleFillerGPU<Geometry>,
-               std::shared_ptr<mpcd::RejectionVirtualParticleFillerGPU<Geometry>>>(
-        m,
-        name.c_str(),
-        py::base<mpcd::RejectionVirtualParticleFiller<Geometry>>())
+               mpcd::RejectionVirtualParticleFiller<Geometry>,
+               std::shared_ptr<mpcd::RejectionVirtualParticleFillerGPU<Geometry>>>(m, name.c_str())
         .def(pybind11::init<std::shared_ptr<SystemDefinition>,
                             const std::string&,
                             Scalar,

--- a/hoomd/mpcd/module.cc
+++ b/hoomd/mpcd/module.cc
@@ -28,6 +28,8 @@ void export_CollisionMethod(pybind11::module&);
 #ifdef ENABLE_MPI
 void export_Communicator(pybind11::module&);
 #endif // ENABLE_MPI
+void export_ConcentricCylindersGeometry(pybind11::module&);
+void export_ConcentricCylindersGeometryFiller(pybind11::module&);
 void export_ConstantForce(pybind11::module&);
 void export_CosineChannelGeometry(pybind11::module&);
 void export_CosineChannelGeometryFiller(pybind11::module&);
@@ -36,8 +38,6 @@ void export_CosineExpansionContractionGeometryFiller(pybind11::module&);
 void export_Integrator(pybind11::module&);
 void export_ManualVirtualParticleFiller(pybind11::module&);
 void export_NoForce(pybind11::module&);
-void export_ConcentricCylindersGeometry(pybind11::module&);
-void export_ConcentricCylindersGeometryFiller(pybind11::module&);
 void export_ParallelPlateGeometry(pybind11::module&);
 void export_ParallelPlateGeometryFiller(pybind11::module&);
 void export_PlanarPoreGeometry(pybind11::module&);
@@ -56,6 +56,7 @@ void export_CellThermoComputeGPU(pybind11::module&);
 #ifdef ENABLE_MPI
 void export_CommunicatorGPU(pybind11::module&);
 #endif // ENABLE_MPI
+void export_ConcentricCylindersGeometryFillerGPU(pybind11::module&);
 void export_CosineChannelGeometryFillerGPU(pybind11::module&);
 void export_CosineExpansionContractionGeometryFillerGPU(pybind11::module&);
 void export_ParallelPlateGeometryFillerGPU(pybind11::module&);
@@ -203,6 +204,8 @@ PYBIND11_MODULE(_mpcd, m)
 #ifdef ENABLE_MPI
     export_Communicator(m);
 #endif // ENABLE_MPI
+    export_ConcentricCylindersGeometry(m);
+    export_ConcentricCylindersGeometryFiller(m);
     export_ConstantForce(m);
     export_CosineChannelGeometry(m);
     export_CosineChannelGeometryFiller(m);
@@ -211,11 +214,8 @@ PYBIND11_MODULE(_mpcd, m)
     export_Integrator(m);
     export_ManualVirtualParticleFiller(m);
     export_NoForce(m);
-    export_ConcentricCylindersGeometry(m);
-    export_ConcentricCylindersGeometryFiller(m);
     export_ParallelPlateGeometry(m);
     export_ParallelPlateGeometryFiller(m);
-    ;
     export_PlanarPoreGeometry(m);
     export_PlanarPoreGeometryFiller(m);
     export_Sorter(m);
@@ -230,6 +230,7 @@ PYBIND11_MODULE(_mpcd, m)
 #ifdef ENABLE_MPI
     export_CommunicatorGPU(m);
 #endif // ENABLE_MPI
+    export_ConcentricCylindersGeometryFillerGPU(m);
     export_CosineChannelGeometryFillerGPU(m);
     export_CosineExpansionContractionGeometryFillerGPU(m);
     export_ParallelPlateGeometryFillerGPU(m);

--- a/hoomd/mpcd/pytest/test_fill.py
+++ b/hoomd/mpcd/pytest/test_fill.py
@@ -22,6 +22,10 @@ def snap():
 @pytest.mark.parametrize(
     "cls, init_args",
     [
+        (hoomd.mpcd.geometry.ConcentricCylinders, {
+            "inner_radius": 4.0,
+            "outer_radius": 8.0
+        }),
         (hoomd.mpcd.geometry.CosineChannel, {
             "amplitude": 4.0,
             "repeat_length": 20.0,
@@ -44,7 +48,7 @@ def snap():
         }),
     ],
     ids=[
-        "CosineChannel", "CosineExpansionContraction", "ParallelPlates",
+        "ConcentricCylinders","CosineChannel", "CosineExpansionContraction", "ParallelPlates",
         "PlanarPore", "Sphere"
     ],
 )

--- a/hoomd/mpcd/pytest/test_fill.py
+++ b/hoomd/mpcd/pytest/test_fill.py
@@ -48,8 +48,8 @@ def snap():
         }),
     ],
     ids=[
-        "ConcentricCylinders","CosineChannel", "CosineExpansionContraction", "ParallelPlates",
-        "PlanarPore", "Sphere"
+        "ConcentricCylinders", "CosineChannel", "CosineExpansionContraction",
+        "ParallelPlates", "PlanarPore", "Sphere"
     ],
 )
 class TestGeometryFiller:

